### PR TITLE
Operator for fixed RNN (CuDNN style)

### PR DESCRIPTION
**Pull request by dzhulgakov**
Pretty much adds a spec for a fixed stacked RNN that can be supported by CuDNN backend. Attributes are taken almost directly from CuDNN/TensorRT spec.

One interesting question is how to standardize the layout of parameters. Theoretically, layout of parameters in CuDNN is opaque (and one needs to set them via separate calls), however we need to choose some. I happen to pick the layout of parameters that corresponds to the real one used by CuDNN (did some reverse engineering). Backend implementations should still go with cudnn calls to reorder the parameters, but it might be convenient for some early testing.

Alternative to merging all parameters in a single tensor would be having directions * layers * 2 * 4 * 2 separate tensors which would be too much in my opinion.

When we land the generic RNN support in ONNX, I plan to add a standard translator from the described format to individual tensors that can be fed into generic RNN operators. That translator can also handle the weights reordering.

Reference cudnn wrappers in frameworks:
- Caffe2: https://github.com/caffe2/caffe2/blob/master/caffe2/operators/recurrent_op_cudnn.h
- PyTorch:
https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/rnn.py
https://github.com/pytorch/pytorch/blob/ceb4f84d12304d03a6a46693e54390869c0c208e/torch/nn/_functions/rnn.py#L254
https://github.com/pytorch/pytorch/blob/625850c2c2f28c67881dc24f5d3abb80fd6e8483/torch/backends/cudnn/__init__.py#L40
https://github.com/pytorch/pytorch/blob/master/torch/backends/cudnn/rnn.py
- CNTK: https://docs.microsoft.com/en-us/cognitive-toolkit/OptimizedRNNStack
- mxnet: https://github.com/apache/incubator-mxnet/blob/master/src/operator/cudnn_rnn-inl.h

### CODE REVIEW

File: onnx/defs/rnn/defs.cc [hunk 1]
```diff
@@ -5,3 +5,114 @@
...
+  `relu`
+  - Ht = ReLU(Wi*Xt + Ri*Ht-1 + Wbi + Rbi)
+  `tanh`
+  - Ht = tanh(Wi*Xt + Ri*Ht-1 + Wbi + Rbi)
+  `lst`
+  - it = sigmoid(Wi*Xt + Ri*Ht-1 + Wbi + Rbi)
+  - ft = sigmoid(Wf*Xt + Rf*Ht-1 + Wbf + Rbf)
+  - ot = sigmoid(Wo*Xt + Ro*Ht-1 + Wbo + Rbo)
+  - ct = tanh(Wc*Xt + Rc*Ht-1 + Wbc + Rbc)
+  - C = f * Ct-1 + it * ct
```
Comment (bddppq) [inline onnx/defs/rnn/defs.cc]: `ft`

File: onnx/defs/rnn/defs.cc [hunk 2]
```diff
@@ -5,3 +5,114 @@
...
+// CuDNN parameters not included yet:
+// - dropout (as we primarily target inference)
+// Description below is borrowed from CuDNN and TensorRT docs
+
+OPERATOR_SCHEMA(OptimizedRNN)
+    .NumInputs(2, 4)
+    .NumOutputs(1, 3)
+    .SetDoc(R"DOC(
+Computes a stack of several RNNs in optimized fashion. This operator corresponds
+to CuDNN functionality and serves as a thin layer on top.
```
Comment (ezyang) [inline onnx/defs/rnn/defs.cc]: ONNX doesn't mandate implementation, right? So I'd reword this as, "This operator is usually implemented via CuDNN"

File: onnx/defs/rnn/defs.cc [hunk 3]
```diff
@@ -5,3 +5,114 @@
...
+
+Equation definitions:
+`i` - input gate
+`o` - output gate
+`f` - forget gate
+`z` - update gate
+`r` - reset gate
+`c` - cell gate
+`h` - hidden gate
+`t` - time step(t-1 means previous time step)
```
Comment (ezyang) [inline onnx/defs/rnn/defs.cc]: Add a space here?
